### PR TITLE
Add padding operator to the PyTorch API

### PR DESCRIPTION
**Pull request by sararb**
### Goals :soccer:
- Add support for padding sequential input data in MM Pytorch backend

### Implementation Details :construction:
- Implement TabularPadding to pad a group of sequential inputs

### Testing Details :mag:

- Add tests of TabularPadding op in `test_sequences.py`

### CODE REVIEW

File: merlin/models/torch/transforms/sequences.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+class TabularPadding(nn.Module):
+    """A PyTorch module for padding tabular sequence data.
+    Parameters
+    ----------
+    schema : Schema
+        The schema of the tabular data, which defines the column names of input features.
+    max_sequence_length : Optional[int], default=None
+        The maximum length of the sequences after padding.
+        If None, sequences will be padded to the maximum length in the current batch.
+    Examples:
```
Comment (marcromeyn) [inline merlin/models/torch/transforms/sequences.py:35]:
```suggestion
    
    Example usage::
```

File: merlin/models/torch/transforms/sequences.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+import torch.nn.functional as F
+from torch import nn
+
+from merlin.models.torch.batch import Batch, Sequence
+from merlin.schema import Schema, Tags
+
+
+class TabularPadding(nn.Module):
+    """A PyTorch module for padding tabular sequence data.
+    Parameters
```
Comment (marcromeyn) [inline merlin/models/torch/transforms/sequences.py:28]:
```suggestion
    
    Parameters
```

File: merlin/models/torch/transforms/sequences.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    ):
+        super().__init__()
+        self.schema = schema
+        self.max_sequence_length = max_sequence_length
+        self.features: List[str] = self.schema.column_names
+        self.sparse_features = self.schema.select_by_tag(Tags.SEQUENCE).column_names
+        self.padding_idx = 0
+
+    def forward(
+        self, inputs: Union[Batch, torch.Tensor, Dict[str, torch.Tensor]], batch: Batch
```
Comment (marcromeyn) [inline merlin/models/torch/transforms/sequences.py:68]: I think a yet-to-be added `BatchBlock` will ensure:

```python
inputs: Union[torch.Tensor, Dict[str, torch.Tensor]]
```

**Review by marcromeyn — APPROVED** [review]
I left some small comments
